triton-vm/src/stark.rs

```diff
@@ -270,38 +278,52 @@ impl Stark {
         );
         let grand_cross_table_arg_quotient_codeword = grand_cross_table_arg
             .terminal_quotient_codeword(
-                &full_fri_domain_tables,
-                &self.fri.domain,
-                derive_omicron(full_fri_domain_tables.padded_height as u64),
+                &full_quotient_domain_tables,
+                &quotient_domain,
+                derive_trace_domain_generator(full_quotient_domain_tables.padded_height as u64),
             );
         quotient_codewords.push(grand_cross_table_arg_quotient_codeword);
 
         let grand_cross_table_arg_quotient_degree_bound = grand_cross_table_arg
             .quotient_degree_bound(
-                &full_fri_domain_tables,
+                &full_quotient_domain_tables,
                 self.parameters.num_trace_randomizers,
             );
         quotient_degree_bounds.push(grand_cross_table_arg_quotient_degree_bound);
         prof_stop!(maybe_profiler, "grand cross table");
 
         prof_start!(maybe_profiler, "nonlinear combination");
+        // magic number `1` corresponds to `num_randomizer_polynomials`, which is currently ignored
+        let (randomizer_weight, base_ext_quot_weights) = non_lin_combi_weights.split_at(1);
         let combination_codeword = self.create_combination_codeword(
-            vec![x_rand_codeword],
-            base_fri_domain_codewords,
-            extension_fri_domain_codewords,
+            &quotient_domain,
+            base_quotient_domain_codewords,
+            extension_quotient_domain_codewords,
             quotient_codewords,
-            non_lin_combi_weights.to_vec(),
+            base_ext_quot_weights.to_vec(),
             base_degree_bounds,
             extension_degree_bounds,
             quotient_degree_bounds,
             maybe_profiler,
         );
+
+        prof_start!(maybe_profiler, "LDE 3");
+        let combination_polynomial = quotient_domain.interpolate(&combination_codeword);
+        let fri_combination_codeword_without_randomizer =
+            self.fri.domain.evaluate(&combination_polynomial);
+        prof_stop!(maybe_profiler, "LDE 3");
+
+        let fri_combination_codeword: Vec<_> = fri_combination_codeword_without_randomizer
+            .into_par_iter()
+            .zip_eq(x_rand_codeword.into_par_iter())
+            .map(|(cc_elem, rand_elem)| cc_elem + randomizer_weight[0] * rand_elem)
```

Comment:
Actually, the randomizer codeword does not need a random weight. If we are going to treat this term specially anyway, I would suggest dropping the weight.

Reply:
You are right. I was confused about this when debugging. I have decided to not change the behavior in this PR because that means that I don't have to change the verifier, which gives me great confidence that introducing the quotient domain does not break anything, like for example soundness.